tests/metrics/test_fidelity.py

```diff
@@ -44,64 +45,48 @@ def test_causal_metrics():
                 assert 0.0 <= score <= 1.0
 
 
-def test_perturbation_metrics():
-    # ensure we can compute insertion/deletion metric with consistent arguments
-    input_shape, nb_labels, nb_samples = ((20, 10), 5, 50)
-    x, y = generate_data(input_shape, nb_labels, nb_samples)
-    model = generate_timeseries_model(input_shape, nb_labels)
-    explanations = np.random.uniform(0, 1, x.shape)
-
-    def inverse(x):
-        maximums = x.max(axis=1)
-        maximums = np.expand_dims(maximums, axis=1)
-        maximums = np.repeat(maximums, x.shape[1], axis=1)
-        baselines = maximums - x
-        return baselines
-
-    for step in [-1, 10]:
-        for baseline_mode in [0.0, inverse]:
-            for metric in ["loss", "accuracy"]:
-                score_insertion = InsertionTS(
-                    model, x, y, metric=metric, baseline_mode=baseline_mode,
-                    steps=step, max_percentage_perturbed=0.2,
-                ).evaluate(explanations)
-                score_deletion = DeletionTS(
-                    model, x, y, metric=metric, baseline_mode=baseline_mode,
-                    steps=step, max_percentage_perturbed=0.2,
-                ).evaluate(explanations)
-
-                for score in [score_insertion, score_deletion]:
-                    if metric == "loss":
-                        assert 0.0 < score
-                    elif metric == "accuracy":
-                        assert 0.0 <= score <= 1.0
-
-
-def test_regression_metrics():
-    # ensure we can compute insertion/deletion metric with consistent arguments
-    input_shape, nb_labels, nb_samples = ((20, 10), 5, 50)
-    x, y = generate_data(input_shape, nb_labels, nb_samples)
-    model = generate_regression_model(input_shape, nb_labels)
-    explanations = np.random.uniform(0, 1, x.shape)
+def test_data_types_and_operators():
+    # ensure that the different metrics are robust to several data types and operators
+    # test several data types
+    data_types = {
+        "tabular": (16,),
+        "time-series": (16, 8),
+        "images": (16, 16, 3)
+    }
+
+    for type, input_shape in data_types.items():
+        nb_labels, samples = 10, 20
+        x, y = generate_data(input_shape, nb_labels, samples)
+        model = generate_regression_model(input_shape, nb_labels)
+
+        explanations = np.random.uniform(0, 1, x.shape[:-1])
+    
+    # test several
```

Comment:
I do not know if its a mistake or not but here you say you will test several operators but only one is in the dictionnary.

Reply:
Initially, there was also the `prediction_operator` however, for regression model, it could output negative values. Hence, fidelity metrics could also output negative values. Which does not correspond to the expected behavior of the metrics.

Hence it is a mistake on my part. Nonetheless, it leads me to the question, should we take the absolute value of the operator output in the metrics.